package/src/package.rs

```diff
@@ -0,0 +1,305 @@
+        match self {
+            Self::Executable(ref prog) => prog.mast_forest(),
+            Self::Library(ref lib) => lib.mast_forest(),
+        }
+    }
+}
+
+// PACKAGE MANIFEST
+// ================================================================================================
+
+/// The manifest of a package, containing the set of package dependencies(libraries or packages) and
+/// exported procedures and their signatures, if known.
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(default)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct PackageManifest {
+    /// The set of exports in this package.
+    pub exports: BTreeSet<PackageExport>,
+    /// The libraries(packages) linked against by this package, which must be provided when
+    /// executing the program.
+    pub dependencies: Vec<Dependency>,
+}
+
+/// A procedure exported by a package, along with its digest and
+/// signature(will be added after MASM type attributes are implemented).
+#[derive(Clone, PartialEq, Eq, Serialize, Deserialize, Ord, PartialOrd)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct PackageExport {
+    /// The fully-qualified name of the procedure exported by this package
+    pub name: String,
+    /// The digest of the procedure exported by this package
+    #[serde(
+        serialize_with = "se::serialize_digest",
+        deserialize_with = "de::deserialize_digest"
+    )]
+    #[cfg_attr(test, proptest(value = "Digest::default()"))]
+    pub digest: Digest,
+    // Signature will be added in the future when the type signatures are available in the Assembly
+    // #[serde(default)]
+    // pub signature: Option<Signature>,
+}
+
+impl fmt::Debug for PackageExport {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("PackageExport")
+            .field("name", &format_args!("{}", self.name))
+            .field("digest", &format_args!("{}", DisplayHex::new(&self.digest.as_bytes())))
+            // .field("signature", &self.signature)
+            .finish()
+    }
+}
+
+// PACKAGE
+// ================================================================================================
+
+/// A package containing a [Program]/[Library], and a manifest(exports and dependencies).
+#[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq)]
+#[cfg_attr(test, derive(proptest_derive::Arbitrary))]
+pub struct Package {
+    /// Name of the package
+    pub name: String,
+    /// The MAST artifact ([Program] or [Library]) of the package
+    #[serde(serialize_with = "se::serialize_mast", deserialize_with = "de::deserialize_mast")]
+    pub mast: MastArtifact,
+    /// The package manifest, containing the set of exported procedures and their signatures,
+    /// if known.
+    pub manifest: PackageManifest,
+}
+
+impl Package {
+    const MAGIC: &'static [u8] = b"MASP\0";
+    const FORMAT_VERSION: &'static [u8] = b"1.0\0";
```

Comment:
nit: could we move those at the top of the file, similar to [how we do with `MastForest`](https://github.com/0xPolygonMiden/miden-vm/blob/ebc8311467833bd34ac3bf0592f857390a37141d/core/src/mast/serialization/mod.rs#L80-L88)? Do we also want the same serialized format for versions as `MastForest` for consistency?

Also maybe for consistency with `MastForest` we'd use version 0 for now?

Reply:
Done.